Sources/ZcashLightClientKit/Synchronizer/SDKSynchronizer.swift

```diff
@@ -652,6 +652,173 @@ public class SDKSynchronizer: Synchronizer {
+        }
+
+        // Parallel part
+        var checkResults: [String: CheckResult] = [:]
+        
+        await withTaskGroup(of: CheckResult.self) { group in
+            for service in services {
+                group.addTask {
+                    let startTime = Date().timeIntervalSince1970
+                    let info = try? await service.service.getInfo()
+                    let markTime = Date().timeIntervalSince1970
+                    let latestBlockHeight = try? await service.service.latestBlockHeight()
+                    let endTime = Date().timeIntervalSince1970
+
+                    let getInfoTime = markTime - startTime
+                    let latestBlockHeightTime = endTime - markTime
+                    let mean = (getInfoTime + latestBlockHeightTime) / 2
+
+                    return CheckResult(
+                        id: service.url,
+                        info: info,
+                        getInfoTime: getInfoTime,
+                        latestBlockHeight: latestBlockHeight,
+                        latestBlockHeightTime: latestBlockHeightTime,
+                        mean: mean,
+                        service: service,
+                        blockTime: 0
+                    )
+                }
+            }
+            
+            var tmpResults: [String: CheckResult] = [:]
+            
+            for await result in group {
+                // rule out results where calls failed
+                guard let info = result.info, let _ = result.latestBlockHeight else {
+                    continue
+                }
+                
+                // rule out if mismatch of networks
+                guard (info.chainName == "main" && network == .mainnet) 
+                    || (info.chainName == "test" && network == .testnet) else {
+                    continue
+                }
+                
+                // rule out mismatch of consensus branch IDs
+                guard let localBranchID = await blockProcessor.consensusBranchIdFor(Int32(info.blockHeight)) else {
+                    continue
+                }
+
+                guard let remoteBranchID = ConsensusBranchID.fromString(info.consensusBranchID) else {
+                    continue
+                }
+                
+                guard remoteBranchID == localBranchID else {
+                    continue
+                }
+
+                // rule out syncing server
+                guard info.blockHeight + UInt64(nBlocks) >= info.estimatedHeight else {
+                    continue
+                }
+                
+                tmpResults[result.id] = result
+            }
+
+            // rule out all means above latencyThreshold
+            let sortedUnderThreshold = tmpResults.compactMap {
+                $0.value.mean < (latencyThresholdMillis / 1000.0) ? $0 : nil
+            }
+
+            // sort the server responses by mean
+            let sortedServers = sortedUnderThreshold.sorted {
```

Comment:
```suggestion
            let sortedUnderThreshold = underThreshold.sorted {
```
Swapping the order of operations was done incorrectly; `sortedServers` was not used. This fixes it.

Reply:
Updated